odoo/models.py

```diff
@@ -1865,9 +1867,11 @@ def _read_group(self, domain, groupby=(), aggregates=(), having=(), offset=0, li
         having_expression, having_params, fnames_used = self._read_group_having(having, query)
         fnames_to_flush.update(fnames_used)
 
-        orderby_terms, extra_groupby_terms, fnames_used = self._read_group_orderby(order, groupby, query)
+        orderby_terms, extra_groupby_terms, fnames_used = self._read_group_orderby(order, groupby_terms, query)
         fnames_to_flush.update(fnames_used)
 
+        groupby_terms = list(groupby_terms.values())
```

Comment:
It is not convenient to reuse a variable and change its type during a method ... makes code hard to follow. Can't you have another name for the dict-based "groupby_terms" ?